resourcehandler/k8sresources.go

```diff
@@ -68,6 +73,31 @@ func (k8sHandler *K8sResourceHandler) GetResources(frameworks []reporthandling.F
 	return k8sResourcesMap, allResources, nil
 }
 
+func getCloudProviderDescription(allResources map[string]workloadinterface.IMetadata, k8sResourcesMap *cautils.K8SResources) error {
+	if cloudsupport.IsRunningInCloudProvider() {
+		wl, err := cloudsupport.GetDescriptiveInfoFromCloudProvider()
+		if err != nil {
+			cluster := k8sinterface.GetCurrentContext().Cluster
+			provider := cloudsupport.GetCloudProvider(cluster)
+			// Return error with useful info on how to configure credentials for getting cloud provider info
+			switch provider {
+			case "gke":
+				return fmt.Errorf("could not get descriptive information about gke cluster: %s using sdk client. See https://developers.google.com/accounts/docs/application-default-credentials for more information", cluster)
+			case "eks":
+				return fmt.Errorf("could not get descriptive information about eks cluster: %s using sdk client. Check out how to configure credentials in https://docs.aws.amazon.com/sdk-for-go/api/", cluster)
+			case "aks":
+				return fmt.Errorf("could not get descriptive information about aks cluster: %s. %v", cluster, err.Error())
+			}
+			return err
+		}
+		allResources[wl.GetID()] = wl
+		// k8sResourcesMap[<ID>] = workload
+		(*k8sResourcesMap)[fmt.Sprintf("%s/%s/%ss", wl.GetApiVersion(), wl.GetNamespace(), strings.ToLower(wl.GetKind()))] = []string{wl.GetID()}
```

Comment:
You should use built in functions for converting single kind to plural.
consider calling [this function](https://github.com/armosec/k8s-interface/blob/6f24b7195a92aa1fb71927bc52c5d124d2afa757/k8sinterface/k8sdiscovery.go#L131)

Reply:
Also, you do not want to use the namespace, only group, version and kind. This are the resources mentioned in the match object

Reply:
I see it returns a list... you want me to use it anyway?